src/PageContent/RequestPickup/RecurringPickupRequest.js

```diff
@@ -0,0 +1,284 @@
+        super(props);
+        this.state = {
+            memberList: [
+                {id: 'dhA03LwTp3cibXVUcb3nQqO34wj1', name: 'Test DA1 Member1'}, 
+                {id: 'fbCm3Yrbi4e12WgpVz3gq25VKea2', name: 'Test DA1 Member2'}
+            ],
+            delivererGroups: [
+                {id: 'R8BAHrxdkfQoAmfWvGa1OJmjQP43', name: 'Test DG1'},
+                {id: 'sS4dqgxgLIXtPf42DydgkWLWeHT2', name: 'Test DG2'}
+            ],
+            receivingAgencies: [
+                {id: 'uCm0OG4WeoSyjk3c0rdY3mlaBXl2', name: 'Test RA2'},
+                {id: 'uGOFJ8NqHjbZhKAYzSZFRs1dSKD3', name: 'Test RA1'}
+            ],
+            fields: {},
+            errors: {},
+            isOpen: false,
+            formInfo: [],
+            dayOfWeek: ''
+        }; 
+        this.submitRequest = this.submitRequest.bind(this);
+        this.createRequest = this.createRequest.bind(this);
+        this.toggleModal = this.toggleModal.bind(this);
+    }
+
+    componentDidMount(){
+        // Query DB and setState for memberList, delivererGroups, and receivingAgencies
+    }
+
+    handleValidation(){
+        let fields = this.state.fields;
+        let errors = {};
+        let formIsValid = true;
+
+        //Date
+        if(!fields['startDate']){
+            formIsValid = false;
+            errors['startDate'] = 'Start date cannot be empty';
+        }
+
+        if((fields['recurTimes'] === '' || !fields['recurTimes']) && fields['endDate'] !== '' && fields['endDate'] < fields['startDate']){
+            formIsValid = false;
+            errors['endBeforeStart'] = 'End date cannot be before start date';
+        }
+
+        //Time
+        if(!fields['startTime']){
+            formIsValid = false;
+            errors['startTime'] = 'Start time cannot be empty';
+        }
+
+        if(!fields['endTime'] && (!fields['recurTimes'] || fields['recurTimes'] === '')){
+            formIsValid = false;
+            errors['endTime'] = 'End time cannot be empty';
+        }
+
+        if(fields['endTime'] < fields['startTime']){
+            formIsValid = false;
+            errors['time'] = 'Invalid time range';
+        }    
+
+        //Recur times
+        if(fields['recurTimes'] <= 0 && fields['endDate'] !== '') {
+            formIsValid = false;
+            errors['recurTimes'] = 'Pickup must recur at least once';
+        }
+
+        if((fields['recurTimes'] === '' && fields['endDate'] === '') 
+            || (!fields['recurTimes'] && !fields['endDate'])){
+            formIsValid = false;
+            errors['recurTimes'] = 'Must enter value for recurring times or an end date';
+        }
+
+        this.setState({errors: errors});
+        return formIsValid;
+    }
+
+    handleChange(field, e){         
+        let fields = this.state.fields;
+        fields[field] = e.target.value;        
+        this.setState({fields});
+    }
+
+    toggleModal(){
+        this.setState({
+            isOpen: !this.state.isOpen
```

Comment:
This will cause problems. For why and how to fix it, read: https://reactjs.org/docs/faq-state.html#why-is-setstate-giving-me-the-wrong-value